### PR TITLE
递归神经网络

### LINKED ISSUE #50: Recurrent Neural Networks
* Markdown文件：[文件地址](https://github.com/xitu/tensorflow/blob/zh-hans/tensorflow/docs_src/tutorials/recurrent.md)
  * PR 地址：#66
  ----
  * 翻译时间：3 天
  * 校对时间：1 天
  * 翻译积分：3 分
  * 校对积分：1 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by edvardHua**
完成翻译 @leviding 
resolve #50

### CODE REVIEW

**Comment by leviding** [comment]
@edvardHua 要标注对应的 issue

**Comment by edvardHua** [comment]
好的。

**Comment by witmax** [comment]
@leviding 校对认领

**Comment by leviding** [comment]
@witmax ok

**Comment by TobiasLee** [comment]
@leviding 校对认领

File: tensorflow/docs_src/tutorials/recurrent.md [hunk 1]
```diff
@@ -183,49 +146,41 @@ stacked_lstm = tf.contrib.rnn.MultiRNNCell(
...
+## 除此之外
 
-There are several tricks that we haven't mentioned that make the model better,
-including:
+还有几个优化模型的技巧没有提到，包括：
 
-* decreasing learning rate schedule,
-* dropout between the LSTM layers.
+* 随时间降低学习率。
+* LSTM 层间 dropout。
```
Comment (TobiasLee) [inline tensorflow/docs_src/tutorials/recurrent.md]: LSTM 层间 dropout。 =>
在 LSTM 层间使用 dropout。

File: tensorflow/docs_src/tutorials/recurrent.md [hunk 2]
```diff
@@ -1,62 +1,40 @@
...
 
-The data required for this tutorial is in the `data/` directory of the
-[PTB dataset from Tomas Mikolov's webpage](http://www.fit.vutbr.cz/~imikolov/rnnlm/simple-examples.tgz).
+本教程需要的数据在 data/ 路径下，来源于 [Tomas Mikolov 网站上的 PTB 数据集](http://www.fit.vutbr.cz/~imikolov/rnnlm/simple-examples.tgz)
 
-The dataset is already preprocessed and contains overall 10000 different words,
-including the end-of-sentence marker and a special symbol (\<unk\>) for rare
-words. In `reader.py`, we convert each word to a unique integer identifier,
-in order to make it easy for the neural network to process the data.
+该数据集已经预先处理过并且包含了全部的 10000 个不同的词语，其中包括语句结束标记符，以及标记稀有词语的特殊符号 (<unk>) 。我们在 `reader.py` 中转换所有的词语，让他们各自有唯一的整型标识符，便于神经网络处理。
```
Comment (TobiasLee) [inline tensorflow/docs_src/tutorials/recurrent.md]: 该数据集已经预先处理过并且包含了全部的 10000 个不同的词语 =>
该数据集已经经过预处理，总共包含了 10000 个不同的词语

File: tensorflow/docs_src/tutorials/recurrent.md [hunk 3]
```diff
@@ -70,110 +48,95 @@ words_in_dataset[4] = [quick, high]
...
 
-By design, the output of a recurrent neural network (RNN) depends on arbitrarily
-distant inputs. Unfortunately, this makes backpropagation computation difficult.
-In order to make the learning process tractable, it is common practice to create
-an "unrolled" version of the network, which contains a fixed number
-(`num_steps`) of LSTM inputs and outputs. The model is then trained on this
-finite approximation of the RNN. This can be implemented by feeding inputs of
-length `num_steps` at a time and performing a backward pass after each
-such input block.
+RNN 的输出结果依赖于不定长度的输入，这是它网络的特点所决定的。不幸的是，这让反向传播的计算变得很困难。为了能够学习流程易于处理，通过的做法是创建一个「unrolled」版本的网络，这个网络包含了固定数量（`num_steps`）的 LSTM 输入和输出。这样模型就可以有限近似 RNN 的形式来训练。这可以每次通过提供长度为 `num_steps` 的输入和每次迭代完成之后进行反向传导。
```
Comment (TobiasLee) [inline tensorflow/docs_src/tutorials/recurrent.md]: 这样模型就可以有限近似 RNN 的形式来训练 =>
这样模型就可以使用近似 RNN 的方式来训练

Comment (TobiasLee) [inline tensorflow/docs_src/tutorials/recurrent.md]: 这可以每次通过提供长度为 `num_steps` 的输入和每次迭代完成之后进行反向传导。 =>
我们可以通过每次提供长度为 `num_steps` 的输入，和每次迭代完成之后进行反向传导来实现它。

File: tensorflow/docs_src/tutorials/recurrent.md [hunk 4]
```diff
@@ -183,49 +146,41 @@ stacked_lstm = tf.contrib.rnn.MultiRNNCell(
...
-## What Next?
+## 除此之外
 
-There are several tricks that we haven't mentioned that make the model better,
-including:
+还有几个优化模型的技巧没有提到，包括：
 
-* decreasing learning rate schedule,
-* dropout between the LSTM layers.
+* 随时间降低学习率。
```
Comment (witmax) [inline tensorflow/docs_src/tutorials/recurrent.md]: 随时间降低学习率。
=>
降低学习率计划表。

可以附加 learning rate schedule 的相关链接，读者可以进一步阅读理解这个新概念
http://www.cs.cmu.edu/~ymiao/pdnntk/lrate.html
https://machinelearningmastery.com/using-learning-rate-schedules-deep-learning-models-python-keras/

**Comment by witmax** [comment]
@edvardHua @leviding 校对完成

**Comment by witmax** [comment]
整体翻译的不错，行文比较流畅，个别一些地方建议修改下可以读起来更加顺畅

**Comment by leviding** [comment]
@edvardHua 抓紧来修改啦

**Comment by edvardHua** [comment]
@leviding 最近上班比较忙耽搁了一下，@TobiasLee 这位大兄弟对「的」，「地」，「得」的用法研究的很是透彻嘛。

**Comment by leviding** [comment]
@TobiasLee 校对的话，希望能够认真的读每一句话，提出更多的建议供译者参考，第二位校对提出的建议应该比第一位少，但是这里来看多太多了。希望下次校对能更认真的琢磨翻译的优化，提出更多的建议。

**Comment by leviding** [comment]
@witmax 积分增加
